config/src/test/java/org/springframework/security/config/http/OAuth2LoginBeanDefinitionParserTests.java

```diff
@@ -0,0 +1,739 @@
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtDecoderFactory;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.security.web.context.HttpRequestResponseHolder;
+import org.springframework.security.web.context.SecurityContextRepository;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.security.oauth2.core.oidc.TestOidcIdTokens.idToken;
+import static org.springframework.security.oauth2.jwt.TestJwts.jwt;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Tests for {@link OAuth2LoginBeanDefinitionParser}.
+ *
+ * @author Ruby Hartono
+ */
+public class OAuth2LoginBeanDefinitionParserTests {
+	private static final String CONFIG_LOCATION_PREFIX = "classpath:org/springframework/security/config/http/OAuth2LoginBeanDefinitionParserTests";
+
+	@Rule
+	public final SpringTestRule spring = new SpringTestRule();
+
+	@Autowired
+	private ClientRegistrationRepository clientRegistrationRepository;
+
+	@Autowired
+	private OAuth2LoginAuthenticationFilter oauth2LoginAuthenticationFilter;
+
+	@Autowired(required = false)
+	private OAuth2AuthorizedClientRepository oauth2AuthorizedClientRepository;
+
+	@Autowired(required = false)
+	private OAuth2AuthorizedClientService oauth2AuthorizedClientService;
+
+	@Autowired
+	SecurityContextRepository securityContextRepository;
+
+	@Autowired(required = false)
+	private AuthenticationSuccessListener authenticationSuccessListener;
+
+	@Autowired(required = false)
+	private OidcAuthorizationCodeAuthenticationProvider oidcAuthorizationCodeAuthenticationProvider;
+
+	@Autowired(required = false)
+	private JwtDecoderFactory<ClientRegistration> jwtDecoderFactory;
+
+	@Autowired(required = false)
+	private AuthorizationRequestRepository<OAuth2AuthorizationRequest> authorizationRequestRepository = new HttpSessionOAuth2AuthorizationRequestRepository();
```

Comment:
the `HttpSessionOAuth2AuthorizationRequestRepository` was not registered as bean. the method `OAuth2LoginBeanDefinitionParser.getOAuth2AuthorizationRequestRepository` only get the bean definition but not registering it. 

does this bean need to be registered? because it is not needed anywhere

Reply:
> does this bean need to be registered?

No, it defaults in `OAuth2AuthorizationRequestRedirectFilter`. However, since you're now mocking it, you need to keep it here. Just remove the default assignment since the mock will be injected.